packages/angular_devkit/build_angular/src/differential-loading/update-webpack-config.ts

```diff
@@ -0,0 +1,132 @@
+    entry: buildModernEntry(config),
+    resolve: buildModernResolve(config),
+  };
+
+  const newConfig = [legacyConfig, modernConfig];
+
+  return newConfig;
+}
+
+function buildLegacyOutput(config: WebpackConfig): object {
+  const filename = config.output.filename.replace('[name]', '[name].legacy');
+
+  return { ...config.output, filename };
+}
+
+function buildModernOutput(config: WebpackConfig): object {
+  const filename = config.output.filename.replace('[name]', '[name].modern');
+
+  return { ...config.output, filename };
+}
+
+function buildModernResolve(config: WebpackConfig): object {
+  const modernResolve = { ...config.resolve };
+  modernResolve.mainFields = ['es2015', 'module', 'browser', 'main'];
+
+  return modernResolve;
+}
+
+function buildLegacyResolve(config: WebpackConfig): object {
+  const modernResolve = { ...config.resolve };
+  modernResolve.mainFields = ['module', 'browser', 'main'];
+
+  return modernResolve;
+}
+
+function buildModernEntry(config: WebpackConfig): object {
+  const entry = config.entry;
+  const polyfills: string[] = entry.polyfills;
+
+  if (!polyfills) {
+    return { ...entry };
+  }
+
+  const tweakPolyfills = (file: string) => {
+
+    // TODO: Do we need an own polyfill file for modern browsers? If yes, where is it created?
+
+    if (file.endsWith('polyfills.ts')) {
+      return file.replace('polyfills.ts', 'polyfills.modern.ts');
+    } else {
+      return file;
+    }
+  };
+
+  const modernPolyfills = polyfills.map(tweakPolyfills);
+  const modernEntry = { ...entry, polyfills: modernPolyfills };
+
+  return modernEntry;
+}
+
+function buildLegacyPlugins(config: WebpackConfig): object {
+  const plugins = config.plugins as WebpackPlugin[];
+  const legacyPlugins = [...plugins];
+
+  const acpIndex = plugins.findIndex(p => p.constructor.name === 'AngularCompilerPlugin');
+  const acp = plugins[acpIndex];
+  const legacyAcp = buildLegacyAcp(acp);
+  legacyPlugins[acpIndex] = legacyAcp;
+
+  return legacyPlugins;
+}
+
+function buildModernPlugins(config: WebpackConfig): object {
+  const plugins = config.plugins as WebpackPlugin[];
+  const modernPlugins = [...plugins];
+
+  const statsPluginIndex = plugins.findIndex(p => p.constructor.name === 'StatsPlugin');
+  if (statsPluginIndex > -1) {
+    const statsPlugin = plugins[statsPluginIndex];
+    statsPlugin.output = 'stats.legacy.json';
+    const modernStatsPlugin = buildModernStatsPlugin(statsPlugin);
+    modernPlugins[statsPluginIndex] = modernStatsPlugin;
+  }
+
+  return modernPlugins;
+}
+
+function buildModernStatsPlugin(statsPlugin: WebpackPlugin): object {
+  const modernStatsPlugin = new statsPlugin.constructor();
+  modernStatsPlugin.output = 'stats.modern.json';
+
+  return modernStatsPlugin;
+}
+
+function buildLegacyAcp(acp: WebpackPlugin): object {
+  const options = acp._options;
+  const modernOptions = { ...options };
+
+  // TODO: Where shall we create this tsconfig with target=es5?
```

Comment:
will we address the TODO in this PR?

Reply:
Question for reviewers: Where shall we create this tsconfig with target=es5? Schematics? On the fly if it does not exist?